android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java

```diff
@@ -0,0 +1,36 @@
+package com.reactnativecommunity.webview;
+
+import android.widget.FrameLayout;
+
+import com.facebook.react.uimanager.ThemedReactContext;
+
+public class RNCWebView extends FrameLayout {
+  private RNCWebViewManager.InternalWebView internalWebView;
+
+  public RNCWebView(ThemedReactContext reactContext) {
+    super(reactContext);
+    setLayoutParams(new FrameLayout.LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
+  }
+
+  public void attachWebView(RNCWebViewManager.InternalWebView webView) {
```

Comment:
> I am assuming you mean in this method - yeah I think we could do something like that to make this call a bit safer.

Yes, that's what I'm referring to: removing existing webviews before adding a new one inside `attachWebView`.

> We could assert and crash if the webview is already attached to another view? i.e

It could be nice to crash in debug builds (so we can catch issues more quickly) and log an error to Sentry in release builds (to reduce impact on users if we forget to do the check somewhere). I'm also open to crashing in all builds as long as we test this thoroughly and try to prevent any crashes from getting to prod. Crashing in all builds has the benefit of making our user experience on debug builds consistent with the user experience on release builds which will help with debugging bug reports related to this issue.

> I think in the places where we do call this method we check to see if there are any existing webviews and remove them.

Are there any times when we wouldn't do that check? So far, I can't think of a time when we'd want to call `addView(internalWebView)` without doing that check first, so it could help to include that in `attachWebView`.

Anyway, I'm open to both solutions, i.e. either the side effect in `attachWebView` or the crash. Let me know what you prefer!

Reply:
> Are there any times when we wouldn't do that check? So far, I can't think of a time when we'd want to call addView(internalWebView) without doing that check first, so it could help to include that in attachWebView.

I think if we include that side-effect in `attachWebView` we could be masking a possible bug and not properly cleaning up the old WebView.

>  Crashing in all builds has the benefit of making our user experience on debug builds consistent with the user experience on release builds which will help with debugging bug reports related to this issue.

I like this as it's predictable in all cases. I'll proceed with this

Reply:
> I like this as it's predictable in all cases. I'll proceed with this

Sounds good! 👍